components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2676,6 +2675,35 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return project;
     }
 
+    /**
+     * Checks if a project can be created, i.e. the current user has the required permissions
+     * to install webhooks for the given repository.
+     */
+    private async canCreateProject(currentUser: User, cloneURL: string) {
+        try {
+            const parsedUrl = RepoURL.parseRepoUrl(cloneURL);
+            const host = parsedUrl?.host;
+            if (!host) {
+                throw Error("Unknown host: " + parsedUrl?.host);
+            }
+            if (host === "github.com" && this.config.githubApp?.enabled) {
+                const availableRepositories = await this.githubAppSupport.getProviderRepositoriesForUser({
+                    user: currentUser,
+                    provider: "github.com",
+                });
+                return availableRepositories.some((r) => r.cloneUrl === cloneURL);
+            } else {
+                return await this.scmService.canInstallWebhook(currentUser, cloneURL);
+
+                // note: the GitHub App based check is not included in the ProjectService due
```

Comment:
To me this is kind of a red flag to not but both things togehter in a SCMService, honestly. Also, most of that functionality seems util functions around HostContext and the likes, maybe we can place them in such a place (without injection), which also solves the injection cycle.

I would be ok to merge this today in favor of turnarounds, if we resolve it tomorrow maybe? :upside_down_face:

Reply:
Let's resolve this in a different context 🙏🏻